Core/src/Testing/System/SystemTestCase.php

```diff
@@ -89,15 +89,21 @@ public static function randId()
      *
      * @param StorageClient $client
      * @param string $bucketName
-     * @param array $options
+     * @param array $options [optional] {
+     *     Configuration options is an array.
+     *
+     *     @type int $retryLimit Maximum number of retries.
+     *           **Defaults to** `8`.
+     * }
      * @return Bucket
      *
      * @experimental
      * @internal
      */
     public static function createBucket(StorageClient $client, $bucketName, array $options = [])
     {
-        $backoff = new ExponentialBackoff(8);
+        $retryLimit = isset($options['retryLimit']) ? $options['retryLimit'] : 8;
```

Comment:
@dwsupplee 's suggestion: Can we stop retry for `BadRequestException` rather than introducing the `retryLimit`?

Reply:
migrated the `resultLimit` check to a check on `BadRequestException` instead